internal/cache/metadata/stat_cache.go

```diff
@@ -176,30 +201,86 @@ func (sc *statCacheBucketView) AddNegativeEntry(objectName string, expiration ti
 	}
 }
 
+func (sc *statCacheBucketView) AddNegativeEntryForFolder(folderName string, expiration time.Time) {
+	name := sc.key(folderName)
+
+	// Insert a negative entry.
+	e := entry{
+		f:          nil,
+		expiration: expiration,
+		key:        name,
+	}
+
+	if _, err := sc.sharedCache.Insert(name, e); err != nil {
+		panic(err)
+	}
+}
+
 func (sc *statCacheBucketView) Erase(objectName string) {
 	name := sc.key(objectName)
 	sc.sharedCache.Erase(name)
 }
 
 func (sc *statCacheBucketView) LookUp(
 	objectName string,
-	now time.Time) (hit bool, m *gcs.MinObject) {
+	now time.Time) (bool, *gcs.MinObject) {
 	// Look up in the LRU cache.
-	value := sc.sharedCache.LookUp(sc.key(objectName))
+	entry := sc.lruCacheLookup(objectName, now)
+	if entry == nil {
+		return false, nil
+	}
+
+	return true, entry.m
+}
+
+func (sc *statCacheBucketView) LookUpFolder(
+	folderName string,
+	now time.Time) (bool, *controlpb.Folder) {
+	// Look up in the LRU cache.
+	entry := sc.lruCacheLookup(folderName, now)
+	if entry == nil {
+		return false, nil
+	}
+
+	return true, entry.f
+}
+
+func (sc *statCacheBucketView) lruCacheLookup(key string, now time.Time) *entry {
```

Comment:
how about returning a bool here which says entry is found or not.

Reply:
I also had same dilemma to return additional bool or not then I thought its redundant information based entry returned caller can find out if cache is a hit or miss.

Reply:
I would prefer the other way. It would also make the caller job simpler.

func (sc *statCacheBucketView) LookUp(
	objectName string,
	now time.Time) (hit bool, m *gcs.MinObject) {
hit, entry := sc.sc.lruCacheLookup(folderName, now)
if hit {
m = entry.m
}
return
}

Reply:
done